slice/internal/controller/admissioncheck_controller.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+)
+
+type AdmissionCheckReconciler struct {
+	client client.Client
+}
+
+var _ reconcile.Reconciler = (*AdmissionCheckReconciler)(nil)
+
+func NewAdmissionCheckReconciler(cl client.Client) *AdmissionCheckReconciler {
+	return &AdmissionCheckReconciler{
+		client: cl,
+	}
+}
+
+// +kubebuilder:rbac:groups=kueue.x-k8s.io,resources=admissionchecks,verbs=get;list;watch
+// +kubebuilder:rbac:groups=kueue.x-k8s.io,resources=admissionchecks/status,verbs=get;update;patch
+
+func (r *AdmissionCheckReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
```

Comment:
We should be watching for updates to Slice, and updating AC state based on the SliceStatus (one of these states)

https://github.com/AI-Hypercomputer/xpk/blob/a2ef280801303b56d4a8f7e6ec22c336e578c52e/slice/api/v1alpha1/slice_types.go#L90-L99

Reply:
As we discussed, we should update only workload admission check status.